src/components/WalletDropdown/TransactionMenu.tsx

```diff
@@ -1,9 +1,167 @@
+`
+
+interface TransactionInformation {
+  title: string
+  transactions: TransactionDetails[]
+}
+
+const TransactionTitle = styled.span`
+  padding-bottom: 8px;
+  padding-top: 20px;
+  padding-left: 12px;
+  padding-right: 12px;
+  font-weight: 600;
+  color: ${({ theme }) => theme.textTertiary};
+`
+
+const TransactionList = ({ transactionInformation }: { transactionInformation: TransactionInformation }) => {
+  const { title, transactions } = transactionInformation
+
+  return (
+    <TransactionListWrapper key={title}>
+      <TransactionTitle>{title}</TransactionTitle>
+      {transactions.map((transactionDetails) => (
+        <TransactionSummary key={transactionDetails.hash} transactionDetails={transactionDetails} />
+      ))}
+    </TransactionListWrapper>
+  )
+}
+
+const getConfirmedTransactions = (confirmedTransactions: Array<TransactionDetails>) => {
+  const now = new Date().getTime()
+
+  const today: Array<TransactionDetails> = []
+  const currentWeek: Array<TransactionDetails> = []
+  const last30Days: Array<TransactionDetails> = []
+  const currentYear: Array<TransactionDetails> = []
+  const yearMap: { [key: string]: Array<TransactionDetails> } = {}
+
+  confirmedTransactions.forEach((transaction) => {
+    const { addedTime } = transaction
+
+    if (isSameDay(now, addedTime)) {
+      today.push(transaction)
+    } else if (isSameWeek(addedTime, now)) {
+      currentWeek.push(transaction)
+    } else if (now - addedTime < THIRTY_DAYS) {
+      last30Days.push(transaction)
+    } else if (isSameYear(addedTime, now)) {
+      currentYear.push(transaction)
+    } else {
+      const year = getYear(addedTime)
+
+      if (!yearMap[year]) {
+        yearMap[year] = [transaction]
+      } else {
+        yearMap[year].push(transaction)
+      }
+    }
+  })
+
+  const transactionGroups: Array<TransactionInformation> = [
+    {
+      title: 'Today',
+      transactions: today,
+    },
+    {
+      title: 'This week',
+      transactions: currentWeek,
+    },
+    {
+      title: 'Past 30 Days',
+      transactions: last30Days,
+    },
+    {
+      title: 'This year',
+      transactions: currentYear,
+    },
+  ]
+
+  const sortedYears = Object.keys(yearMap)
+    .sort((a, b) => parseInt(b) - parseInt(a))
+    .map((year) => ({ title: year, transactions: yearMap[year] }))
+
+  transactionGroups.push(...sortedYears)
+
+  return transactionGroups.filter((transactionInformation) => transactionInformation.transactions.length > 0)
+}
+
+const EmptyTransaction = styled.div`
+  text-align: center;
+  margin-top: 24px;
+  font-weight: 400;
+  font-size: 14px;
+  padding-left: 12px;
+  padding-right: 12px;
+  color: ${({ theme }) => theme.textSecondary};
+`
+
+export const TransactionHistoryMenu = ({ onClose }: { onClose: () => void }) => {
+  const allTransactions = useAllTransactions()
+  const { chainId } = useWeb3React()
+  const dispatch = useAppDispatch()
+  const transactionGroupsInformation = []
```

Comment:
not *suuuper* important, but I'd put the building of this array in a `useMemo` and move its declaration down to the bottom where you've got the `if` logic happening on line 144